### ISSUE TITLE
TIFF files are invalid as (new) segmentation values

### ISSUE THREAD

**Issue by chrisvanrun** [comic/grand-challenge.org#2672]
Validation always fails for TIFF files when uploading as a component-interface value with sub kind segmentation. (When `overlay_segments` are set).

Creating new segmentation component interfaces forces the addition of the `overlay_segments`. However, TIFF files have no value range calculated as their typical size (~4.000.000.000 voxels) makes it impractical to calculate v.s. regular MHA files (~200.000.000) as such the `image.segments` is never filled and the validation will always fail.

**Comment by chrisvanrun** [comment]
It should be investigated if panimg can retrieve the values present quickly or not.

**Comment by miriam-groeneveld** [comment]
It all depends on the image. We usually deal with multi-resolution images, but that is not always the case. A tiff file can also be a single resolution, single page tiff file of 4 GB. Loading that and getting the unique voxel values from such an image requires a lot of memory. We do extract the min/max value from a tiff file, so maybe that's a good compromise? If you want a segmentation in a tiff file, the overlag segments need to have consecutive values defined, and we just validate on the min/max.

**Comment by jmsmkn** [comment]
Good idea, for that check we would also need to make sure:

- The overlay segments that are defined for the interface are contiguous (e.g. {1, 2, 3} is OK for that check, {10, 20, 30} is not)
- The min of the tiff file is >= the min of the expected overlay segments
- The max of the tiff file is <= the max of the expected overlay segments
- The data type of the tiff file is `int` so that only the integers between the defined min and max value are possible

If that is possible then that should work for most cases.

**Comment by miriam-groeneveld** [comment]
Yes, I think that is the most efficient solution, and a lot of segmentations have contiguous values anyway. Should I make a pitch or do you think it's too little work to justify a pitch?

**Comment by jmsmkn** [comment]
Famous last words but I don't think it's that much work so a PR would be fine.

**Comment by jmsmkn** [comment]
We should check the existing interfaces to see if there are any non contiguous overlays, if not then I think that it would be fine to add as a constraint.

**Comment by miriam-groeneveld** [comment]
There are overlay segments now that are not contiguous.

**Comment by jmsmkn** [comment]
OK then yes we would not be able to validate TIFF files for those ones, so raise a validation error there.
